packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -0,0 +1,422 @@
+}
+
+export interface PolymarketWithEventData extends PolymarketMarketWithAncillaryData {
+  txHash: string;
+  requester: string;
+  proposer: string;
+  timestamp: number;
+  expirationTimestamp: number;
+  eventTimestamp: number;
+  eventBlockNumber: number;
+  eventIndex: number;
+  proposalTimestamp: number;
+  identifier: string;
+  ancillaryData: string;
+  proposedPrice: string;
+}
+
+interface PolymarketOrderBook {
+  market: string;
+  asset_id: string;
+  bids: { price: string; size: string }[];
+  asks: { price: string; size: string }[];
+  hash: string;
+}
+
+export type OrderBookSide = { price: number; size: number }[];
+export interface MarketOrderbooks {
+  orderBooks: [{ bids: OrderBookSide; asks: OrderBookSide }, { bids: OrderBookSide; asks: OrderBookSide }];
+}
+
+export interface StoredNotifiedProposal {
+  txHash: string;
+  question: string;
+  proposedPrice: string;
+  notificationTimestamp: number;
+}
+
+// Helper function to process markets in chunks
+const processMarketsInChunks = async (
+  markets: PolymarketWithEventData[],
+  chunkSize: number,
+  callback: (m: PolymarketWithEventData[]) => Promise<void>
+) => {
+  for (let i = 0; i < markets.length; i += chunkSize) {
+    const chunk = markets.slice(i, i + chunkSize);
+    await callback(chunk);
+    await sleep(500); // Introduce a 0.5 second delay between each chunk
+  }
+};
+
+export const formatPriceEvents = async (
+  events: ProposePriceEvent[]
+): Promise<
+  {
+    txHash: string;
+    requester: string;
+    proposer: string;
+    timestamp: number;
+    eventTimestamp: number;
+    eventBlockNumber: number;
+    expirationTimestamp: number;
+    proposalTimestamp: number;
+    identifier: string;
+    ancillaryData: string;
+    proposedPrice: string;
+    eventIndex: number;
+  }[]
+> => {
+  const ooDefaultLiveness = 7200;
+  return Promise.all(
+    events.map(async (event: ProposePriceEvent) => {
+      const block = await event.getBlock();
+      return {
+        txHash: event.transactionHash,
+        requester: event.args.requester,
+        proposer: event.args.proposer,
+        timestamp: event.args.timestamp.toNumber(),
+        eventTimestamp: block.timestamp,
+        eventBlockNumber: event.blockNumber,
+        expirationTimestamp: event.args.expirationTimestamp.toNumber(),
+        proposalTimestamp: event.args.expirationTimestamp.toNumber() - ooDefaultLiveness,
+        identifier: event.args.identifier,
+        ancillaryData: event.args.ancillaryData,
+        proposedPrice: ethers.utils.formatEther(event.args.proposedPrice),
+        eventIndex: event.logIndex,
+      };
+    })
+  );
+};
+
+export const getPolymarketMarkets = async (params: MonitoringParams): Promise<PolymarketMarket[]> => {
+  const whereClause =
+    " created_at > '2023-01-01'" +
+    " AND question_ID IS NOT NULL" +
+    " AND clob_Token_Ids IS NOT NULL" +
+    ` AND (resolved_by = '${params.binaryAdapterAddress}' OR resolved_by = '${params.ctfAdapterAddress}')`;
```

Comment:
I worry a little about this filter causing us to miss markets if PM upgrades their contract (even a minor update that doesn't affect their ABI). Do you think there might be a way around explicitly filtering for this? We could have a default ABI or we could not filter, but error if we get results that don't match one of these contracts. Thoughts?

Reply:
I think that condition only filters out 1 or 2 old markets that are not used but you are right, this is weak point. I did it to solve [decoding errors](https://github.com/UMAprotocol/protocol/blob/md0x/polymarket-ts-orderbook/packages/monitor-v2/src/monitor-polymarket/common.ts#L199) for these 2 or 3 markets that are not used but I should handle it differently! I will improve this

Reply:
Sounds good! If those markets are unused, we can just have a filter that explicitly removes them to avoid having to write specialized code to deal with them.